flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/context/DefaultContext.java

```diff
@@ -55,15 +58,31 @@ public DefaultContext(Configuration flinkConfig, List<CustomCommandLine> command
                 flinkConfig, PluginUtils.createPluginManagerFromRootFolder(flinkConfig));
 
         Options commandLineOptions = collectCommandLineOptions(commandLines);
+
+        final List<URL> dependencies = new ArrayList<>();
+        // add python dependencies by default
+        try {
+            URL location =
+                    Class.forName(
+                                    "org.apache.flink.python.PythonFunctionRunner",
+                                    false,
+                                    Thread.currentThread().getContextClassLoader())
+                            .getProtectionDomain()
+                            .getCodeSource()
+                            .getLocation();
+            if (Paths.get(location.toURI()).toFile().isFile()) {
+                dependencies.add(location);
+            }
+        } catch (URISyntaxException | ClassNotFoundException e) {
+            throw new SqlExecutionException("Failed to find flink-python jar.", e);
```

Comment:
Can we move this code block to `DefaultContext#load`? I think we will introduce -l/-j comand line paramters in the future.

Reply:
If it is the -j approach, I guess that you want to make the loading of python jar an optional behavior?

Reply:
I think we can modify the constructor of the DefaultContext to pass the dependencies. During the loading, we always try to find the python dependencies and add it into the dependencies.